src/main/scala/viper/silver/plugin/standard/inline/InlineRewrite.scala

```diff
@@ -0,0 +1,84 @@
+package viper.silver.plugin.standard.inline
+
+import scala.collection.mutable
+import viper.silver.ast._
+import viper.silver.ast.utility.ViperStrategy
+
+trait InlineRewrite {
+
+  /**
+    * Rewrites a method by expanding the predicates in its pre- and postconditions,
+    * then by removing the corresponding unfold and fold statements in the body.
+    * Note that we only remove unfolds for predicates from the preconditions,
+    * and folds for predicates from the postconditions.
+    * (I think that's how it works.)
+    */
+  def rewriteMethod(method: Method, program: Program): Method = {
+    val (expandedPres, expandablePrePredsIds) = method.pres
+      .map { expandPredicates(_, method, program) }
+      .unzip
+    val (expandedPosts, expandablePostPredsIds) = method.posts
+      .map { expandPredicates(_, method, program) }
+      .unzip
+    val expandablePrePredIds = expandablePrePredsIds.fold(Set())(_ ++ _)
+    val expandablePostPredIds = expandablePostPredsIds.fold(Set())(_ ++ _)
+    val rewrittenBody = method.body
+      .map { removeFoldUnfolds(_, expandablePrePredIds, expandablePostPredIds) }
+    method.copy(name = method.name,
+        formalArgs = method.formalArgs,
+        formalReturns = method.formalReturns,
+        pres = expandedPres,
+        posts = expandedPosts,
+        body = rewrittenBody
+      )(pos = method.pos, info = method.info, errT = method.errT)
+  }
+
+  /**
+    * Expands all predicates to their bodies in given expression.
+    * 
+    * TODO: Expand only some specified predicates, not all
+    *
+    * @param expr The expression whose predicates will be expanded
+    * @param method The method containing the expression, used to determine locally-scoped variables
+    * @param program The program containing the expression, used to expand predicates
+    * @return The expression with expanded predicates and a set of their names
+    */
+  private[this] def expandPredicates(expr: Exp, method: Method, program: Program): (Exp, Set[String]) = {
+    // Forgive me deities of functional programming for I sin
+    val expandablePredicates = mutable.Set[String]()
+    val expandedExpr = ViperStrategy.CustomContext[Set[String]]({
+      case (expr, args) => expr match {
+        case pred: PredicateAccess =>
```

Comment:
I wonder if this has to be `PredicateAccessPredicate`

Exploratory `println` work showed me that the 
```
requires somePredicate(this)
```
line of Viper gets seen as 
```
requires acc(somePredicate(this, write))
```
by the plugin. This matches with `PredicateAccessPredicate`, not `PredicateAccess` 🤔